include/odp/api/spec/atomic.h

```diff
@@ -176,6 +178,20 @@ void odp_atomic_dec_u32(odp_atomic_u32_t *atom);
  */
 void odp_atomic_max_u32(odp_atomic_u32_t *atom, uint32_t new_max);
 
+/**
+ * Fetch and update maximum value of atomic uint32 variable
+ *
+ * Compares value of atomic variable to the new maximum value. If the new value
+ * is greater than the current value, writes the new value into the variable.
+ * Returns the value of the variable before update operation.
```

Comment:
Suggesting to change: "Returns the value of the variable before update operation." -> "Always returns the value used in the comparison."

It would highlight that the "fetched" value is always returned.

Reply:
That sounds more cumbersome and difficult to understand to me. I do not quite see what the potential misunderstanding in the current wording might be.

Reply:
Maybe "returns the original value of the variable" would be ok if the "before update operation" wording is deemed confusing?

Reply:
```Returns the original value of the variable used in the comparison.``` or
```Returns the original value of the atomic variable.```?

Reply:
Original value is good choice. Remember to update also the return value comment.

Reply:
Updated in V6.

Reply:
Further tweaking in V7.